eva-tools/src/main/java/embl/ebi/variation/eva/vcfDump/VariantExporter.java

```diff
@@ -137,12 +159,202 @@ public VariantExporter(CellBaseClient cellbaseClient) {
+        }
+
+        return regions;
+    }
+
+    private List<Region> getRegionsListForQueryRegions(List<Region> regionsFromQuery) {
+        List<Region> regions = new ArrayList<>();
+        for (Region region : regionsFromQuery) {
+            regions.addAll(regionDivider.divideRegionInChunks(region));
+        }
+        return regions;
+    }
+
+    private QueryOptions addChromosomeSortAndLimitToQuery(String chromosome, boolean ascending) {
+        QueryOptions chromosomeSortedByStartQuery = new QueryOptions(options);
+        chromosomeSortedByStartQuery.put(VariantDBAdaptor.CHROMOSOME, chromosome);
+
+        BasicDBObject sortDBObject = new BasicDBObject();
+        int orderOperator = ascending ? 1 : -1;
+        sortDBObject.put("chr", orderOperator);
+        sortDBObject.put("start", orderOperator);
+        chromosomeSortedByStartQuery.put("sort", sortDBObject);
+
+        chromosomeSortedByStartQuery.put("limit", 1);
+
+        return chromosomeSortedByStartQuery;
+    }
+
+    private int getVariantStart(QueryOptions query) {
+        int start = -1;
+        VariantDBIterator variantDBIterator = variantDBAdaptor.iterator(query);
+        if (variantDBIterator.hasNext()) {
+            Variant variant = variantDBIterator.next();
+            start = variant.getStart();
+        }
+
+        return start;
+    }
+
+
+    private Map<String, List<VariantContext>> dumpVariantsInRegion(Region region, Map<String, VariantSource> sources) throws IOException {
+        Map<String, List<VariantContext>> dumpedVariants = new HashMap<>();
+        QueryOptions regionQuery = addRegionToQuery(region);
+        regionQuery = addProjectionToQuery(regionQuery);
+        VariantDBIterator variantDBIterator = variantDBAdaptor.iterator(regionQuery);
+        while (variantDBIterator.hasNext()) {
+            Variant variant = variantDBIterator.next();
+            if (region.contains(variant.getChromosome(), variant.getStart())) {
+                try {
+                    Map<String, VariantContext> variantContexts = convertBiodataVariantToVariantContext(variant, sources, region);
+                    for (Map.Entry<String, VariantContext> variantContext : variantContexts.entrySet()) {
+                        dumpedVariants.putIfAbsent(variantContext.getKey(), new ArrayList<>());
+                        dumpedVariants.get(variantContext.getKey()).add(variantContext.getValue());
+                    }
+                } catch (Exception e) {
+                    logger.warn("Variant {}:{}:{}>{} dump failed: {}", variant.getChromosome(), variant.getStart(), variant.getReference(),
+                            variant.getAlternate(), e.getMessage());
+                    failedVariants++;
+                }
+                writtenVariants++;
+                if (writtenVariants % 100000 == 0) {
+                    logger.info("written variants: " + writtenVariants);
+                }
```

Comment:
Looks like this `writtenVariants` block should be executed only when no exception is thrown.